README.md

```diff
@@ -92,3 +74,132 @@ some common build and testing tasks. Most of these scripts are implemented with
 - **just-k3d** helpers for interacting with k3d clusters used for testing
 - **just-md** lints markdown in a repository
 - **just-sh** lints shell scripts in a respository.
+
+```yaml
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: linkerd/dev/actions/setup-tools@v35
+      - uses: actions/checkout@v3
+      - run: just-sh lint
+      - run: just-dev lint-actions
+      - run: just-dev check-action-images
+```
+
+### Go
+
+The `go` container provides a Go toolchain (linters and other Go tooling are
+included with the Tools image). The container can be used as a base image when
+building via docker, or it can be used directly in Github like:
+
+```yaml
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container: ghcr.io/linkerd/dev:v35-go
+    steps:
+      - uses: actions/checkout@v3
+      - run: go mod download
+      - run: go test ./...
+```
+
+```yaml
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        k8s:
+          - v1.21
+          - v1.25
+    env:
+      K8S_VERSION: ${{ matrix.k8s }}
+    steps:
+      # Install just* tooling and Go linters
+      - uses: linkerd/dev/actions/setup-tools@v35
+      # Configure the default Go toolchain
+      - uses: linkerd/dev/actions/setup-go@v35
+      - uses: actions/checkout@v3
+      - run: just-k3d create
+      - run: go mod download
+      - run: go test ./...
+```
+
+### Rust
+
+The `rust` container provides a Rust toolchain and associated utilities needed
+to build and test Rust code. The `rust-musl` container provides the same
+toolchain, but with the `musl` target installed with cross-compilation support
+for arm64 and armv7.
+
+These containers can be used in a workflow like so:
+
+```yaml
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container: ghcr.io/linkerd/dev:v35-rust
+    steps:
+      - uses: actions/checkout@v3
+      - run: just-cargo fetch
+      - run: just-cargo test-build
+      - run: just-cargo test
+```
+
+Or, to build a static binary:
+
+```yaml
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container: ghcr.io/linkerd/dev:v35-rust-musl
+    steps:
+      - uses: actions/checkout@v3
+      - run: just-cargo fetch
+      - run: just-cargo build
+```
+
+Build failures are automatically reported as annotations in the GitHub UI.
+
+#### `actions/setup-rust`
+
+The above example is fine when we're just building software, but when we want to
+start networked services in Docker (namely, k3d clusters), the containerized
+approach no longer works.
```

Comment:
Paragraph in wrong section?

Reply:
This is describing how the setup action is needed when the rust container isn't used directly.